hypervisor/src/kvm/mod.rs

```diff
@@ -492,6 +493,37 @@ impl KvmVm {
     pub fn check_extension(&self, c: Cap) -> bool {
         self.fd.check_extension(c)
     }
+
+    #[cfg(target_arch = "x86_64")]
+    /// Translates the MSI extended destination ID bits according to the logic
+    /// found in the Linux kernel's KVM MSI handling.
```

Comment:
Can you add a pointer to this kernel code? Apparently, the vt-d spec only explains how hardware will calculate the address, which is not how system software (e.g. a VMM) should prepare these addresses.